spec/enumerable_spec.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require_relative '../my_enumerable_module.rb'
+
+# rubocop: disable Metrics/BlockLength, Layout/LineLength
+
+describe Enumerable do
+  arr = [5, 6, 7]
+  describe '#my_each' do
+    it 'Returns the same array if block is given' do
+      expect(arr.my_each { |x| x }).to eql(arr)
+    end
+    it 'Returns an enumerable if block is not given' do
+      expect(arr.my_each.class).to eql(Enumerator)
+    end
+  end
+  describe '#my_each_with_index' do
```

Comment:
* [ ] One key functionality of my_each_with_index is its ability to return the index of current value. It will be great to have that functionality tested.

Reply:
Hi Forison, thanks for your feedback!
We didn't add that test case because this method doesn't return the index, it returns the same array we are calling the method on. Instead, it yields both the value and the index to the block, however, we don't have a way to test what is being yielded, here's a screenshot of an example
![Screenshot from 2020-01-28 18-10-28](https://user-images.githubusercontent.com/44302480/73316733-88712500-41f9-11ea-81e3-04a16ce071b6.png)